Source/CarthageKit/Git.swift

```diff
@@ -394,6 +394,10 @@ public func resolveReferenceInRepository(repositoryFileURL: NSURL, reference: St
 /// Attempts to determine whether the given directory represents a Git
 /// repository.
 public func isGitRepository(directoryURL: NSURL) -> SignalProducer<Bool, NoError> {
+	if !NSFileManager.defaultManager().fileExistsAtPath(directoryURL.path!) {
+		return SignalProducer(value: false)
+	}
+
```

Comment:
Instead of adding the check here, we should do it in [`addSubmoduleToRepository()`](https://github.com/Carthage/Carthage/blob/561c0851ffe41594350138bb7c09cd2543913ddf/Source/CarthageKit/Git.swift#L413) I think, because the other use cases are ensured that the given URL exists ([here](https://github.com/Carthage/Carthage/blob/561c0851ffe41594350138bb7c09cd2543913ddf/Source/CarthageKit/Git.swift#L453-L464) and [here](https://github.com/Carthage/Carthage/blob/561c0851ffe41594350138bb7c09cd2543913ddf/Source/CarthageKit/Project.swift#L712-L717)).